src/components/DatePickerDocked/DatePickerDockedHeader.tsx

```diff
@@ -44,8 +45,7 @@ function DatePickerDockedHeader({
 
     const handlePressDropDown = useCallback(
         (type: 'month' | 'year') => {
-            setStore(prev => ({
-                ...prev,
+            setStore(() => ({
                 pickerType: pickerType ? undefined : type,
             }));
         },
```

Comment:
the value for pickerType is available in the function you passed to setStore as the argument.
pickerType will call creating the function everytime.

In this place, it's not affecting. but in other cases it might.

Reply:
resolved @shashank1010